packages/components/src/Table/Table.tsx

```diff
@@ -56,6 +57,14 @@ export type TableHeaderRowProps = {
 const ratioToPercent = (width?: number): string | number | undefined =>
   width != null ? `${width * 100}%` : width
 
+export type TableHeaderRowCellCheckboxProps = {
```

Comment:
Given how many checkbox related type there were, made sense to group these together. I've still left the `checkboxLabel` as optional though so as to provide a more accessible approach but not throw errors in existing implementations